velox/expression/Expr.cpp

```diff
@@ -1372,17 +1369,8 @@ bool Expr::applyFunctionWithPeeling(
         setPeeledArg(leaf, i, numArgs, maybePeeled);
         continue;
       }
-      if ((numLevels == 0 && leaf->isConstant(rows)) ||
-          leaf->isConstantEncoding()) {
-        if (leaf->isConstantEncoding()) {
-          setPeeledArg(leaf, i, numArgs, maybePeeled);
-        } else {
-          setPeeledArg(
-              BaseVector::wrapInConstant(leaf->size(), rows.begin(), leaf),
-              i,
-              numArgs,
-              maybePeeled);
-        }
+      if (leaf->isConstantEncoding()) {
+        setPeeledArg(leaf, i, numArgs, maybePeeled);
```

Comment:
I wonder if we still need the code in https://github.com/mbasmanova/velox-1/blob/1e88c9fa96e6ca3acebd2bfbd9fef47a5a0f4977/velox/expression/Expr.cpp#L1454-L1465

The only reason we need it in the first place was because we were wrapping the vector in an additional const layer of size leaf->size() if leaf->isConstant(rows) was true. however from now on, we only consider vectors to be constant after removing any dictionary layers from their top, which means that the exposed const layer should have the right size to cover all indices from the dict layers above it.